cmd/k8s-operator/deploy/crds/tailscale.com_tsrecorders.yaml

```diff
@@ -1566,15 +1566,21 @@ spec:
                             Bucket name to write to. The bucket is expected to be used solely for
                             recordings, as there is no stable prefix for written object names.
                           type: string
-                        credentialsSecret:
+                        credentials:
                           description: |-
-                            The name of a Kubernetes Secret in the operator's namespace that contains
-                            credentials for writing to the configured bucket. Each key-value pair
-                            from the secret's data will be mounted as an environment variable. It
-                            should include keys for AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY if
-                            using a static access key, or otherwise it will try to acquire credentials
-                            from the file system or the STS API.
-                          type: string
+                            Configure environment variable credentials for managing objects in the
+                            configured bucket. If not set, tsrecorder will try to acquire credentials
+                            first from the file system and then the STS API.
+                          type: object
+                          properties:
+                            secret:
+                              description: |-
+                                The name of a Kubernetes Secret in the operator's namespace that contains
+                                credentials for writing to the configured bucket. Each key-value pair
+                                from the secret's data will be mounted as an environment variable. It
+                                should include keys for AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY if
+                                using a static access key.
+                              type: string
```

Comment:
Alternatively, we could also make `secret` a struc, so that if in future we need to support reading different key names from there (i.e bc the `Secret` was populated by some other tool), we can put those fields under the same struct, i.e

``
secret:
  name: foo
  accessKeyFromKey: aws_access_key_id
  secretAccessKeyFromKey: aws_secret_access_key
```

Reply:
Oh I see - yeah that makes sense. Do you think it's worth keeping the `credentials` layer, or do you think it would make sense to just pull the `secret` up a layer, so e.g. `.spec.storage.s3.secret.name`?

Reply:
To me `.credentials.secret` nesting makes sense 👍🏼